### ISSUE TITLE
“Record with host already exists“

### ISSUE THREAD

**Issue by arnestoe** [brenner-tobias/addon-cloudflared#129]
Hi,
I set up the cloudflare addon newly and that process seemed to work fine. However, it is not working and the log shows

Failed to add route: code: 1003, reason: An A, AAAA, or CNAME record with that host already exists.
[20:32:01] FATAL: Failed to create DNS entry aaaaaaaa.ml

Could you please advise.
Thanks a lot
Arne

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/00-banner.sh
-----------------------------------------------------------
 Add-on: Cloudflared
 Use a Cloudflared tunnel (formerly Argo Tunnel) to remotely connect to Home Assistant without opening any ports
-----------------------------------------------------------
 Add-on version: 2.0.4
 You are running the latest version of this add-on.
 System: Home Assistant OS 8.2  (armv7 / raspberrypi3)
 Home Assistant Core: 2022.7.6
 Home Assistant Supervisor: 2022.07.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
cont-init: info: /etc/cont-init.d/00-banner.sh exited 0
cont-init: info: running /etc/cont-init.d/01-log-level.sh
cont-init: info: /etc/cont-init.d/01-log-level.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-cloudflared-log: starting
s6-rc: info: service init-cloudflared-log successfully started
s6-rc: info: service init-cloudflared-config: starting
[20:31:51] INFO: Checking Add-on config...
[20:31:54] INFO: Checking for existing certificate...
[20:31:54] INFO: Existing certificate found
[20:31:54] INFO: Checking for existing tunnel...
[20:31:54] INFO: Existing tunnel with ID d076ee66-9b54-4ba8-aebe-919d12e02884 found
[20:31:54] INFO: Checking if existing tunnel matches name given in config
[20:31:55] INFO: Existing Cloudflare tunnnel name matches config, proceeding with existing tunnel file
[20:31:56] INFO: Creating config file...
[20:31:59] INFO: Validating config file...
Validating rules from /tmp/config.json
OK
[20:31:59] INFO: Creating new DNS entry aaaaaaaa.ml...
Failed to add route: code: 1003, reason: An A, AAAA, or CNAME record with that host already exists.
[20:32:01] FATAL: Failed to create DNS entry aaaaaaaaa.ml
s6-rc: warning: unable to start service init-cloudflared-config: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
/run/s6/basedir/scripts/rc.init: fatal: stopping the container.
s6-rc: info: service init-cloudflared-log: stopping
s6-rc: info: service init-cloudflared-log successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

**Comment by elcajon** [comment]
```
Failed to add route: code: 1003, reason: An A, AAAA, or CNAME record with that host already exists.
```

Check your Cloudflare DNS settings for an existing entry for your domain to be used with this add-on (`aaaaaaaa.ml`).

**Comment by arnestoe** [comment]
Thanks a lot for the guidance!
Indeed, there was an existing entry. Heven’t put in anything manually, but after deleted the entry the error in the log is gone :)
Unfortunately, it still doesn‘t work. 
Really appreciate additional advise. 
If I open enter my domain name in the browser, I get a 404 error and the log shows:

INF abort: rpc: aborted by remote: rpc: shutdown
2022-07-25T11:46:03Z INF Unregistered tunnel connection connIndex=3
2022-07-25T11:46:03Z WRN Failed to serve quic connection error="failed to accept QUIC stream: Application error 0x0" connIndex=3 ip=198.41.192.57
2022-07-25T11:46:03Z WRN Serve tunnel error error="failed to accept QUIC stream: Application error 0x0" connIndex=3 ip=198.41.192.57




cont-init: info: /etc/cont-init.d/00-banner.sh exited 0
cont-init: info: running /etc/cont-init.d/01-log-level.sh
cont-init: info: /etc/cont-init.d/01-log-level.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-cloudflared-log: starting
s6-rc: info: service init-cloudflared-log successfully started
s6-rc: info: service init-cloudflared-config: starting
[20:57:28] INFO: Checking Add-on config...
[20:57:31] INFO: Checking for existing certificate...
[20:57:31] INFO: Existing certificate found
[20:57:31] INFO: Checking for existing tunnel...
[20:57:31] INFO: Existing tunnel with ID d076ee66-9b54-4ba8-aebe-919d12e02884 found
[20:57:31] INFO: Checking if existing tunnel matches name given in config
[20:57:32] INFO: Existing Cloudflare tunnnel name matches config, proceeding with existing tunnel file
[20:57:33] INFO: Creating config file...
[20:57:36] INFO: Validating config file...
Validating rules from /tmp/config.json
OK
[20:57:37] INFO: Creating new DNS entry Garagen.ml...
2022-07-24T18:57:38Z INF Added CNAME garagen.ml which will route to this tunnel tunnelID=d076ee66-9b54-4ba8-aebe-919d12e02884
[20:57:38] INFO: Finished setting-up the Cloudflare tunnel
s6-rc: info: service init-cloudflared-config successfully started
s6-rc: info: service cloudflared: starting
s6-rc: info: service cloudflared successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[20:57:40] INFO: Connecting Cloudflared Tunnel...
2022-07-24T18:57:40Z INF Starting tunnel tunnelID=d076ee66-9b54-4ba8-aebe-919d12e02884
2022-07-24T18:57:40Z INF Version 2022.7.1
2022-07-24T18:57:40Z INF GOOS: linux, GOVersion: go1.17.10, GoArch: arm
2022-07-24T18:57:40Z INF Settings: map[config:/tmp/config.json cred-file:/data/tunnel.json credentials-file:/data/tunnel.json loglevel:info no-autoupdate:true origincert:/data/cert.pem]
2022-07-24T18:57:40Z INF Generated Connector ID: 7110539e-09aa-47c2-84d6-35713e413f52
2022-07-24T18:57:40Z INF Initial protocol quic
2022-07-24T18:57:40Z INF Starting metrics server on 127.0.0.1:35557/metrics
2022-07-24T18:57:41Z INF Connection 9a7d4792-14c2-44cb-9a27-300d9ea7e6bf registered connIndex=0 ip=198.41.200.193 location=AMS
2022-07-24T18:57:42Z INF Connection ffd8dc1a-655c-4911-974a-792aeaff4c45 registered connIndex=1 ip=198.41.192.107 location=FRA
2022-07-24T18:57:43Z INF Connection 31d01fa5-5d6f-4787-8e64-213560cfdcec registered connIndex=2 ip=198.41.200.43 location=AMS
2022-07-24T18:57:43Z INF Connection 822e94f0-bddc-496d-8cc0-3fe0c60c62d4 registered connIndex=3 ip=198.41.192.57 location=FRA
2022-07-25T11:46:03Z INF abort: rpc: aborted by remote: rpc: shutdown
2022-07-25T11:46:03Z INF Unregistered tunnel connection connIndex=3
2022-07-25T11:46:03Z WRN Failed to serve quic connection error="failed to accept QUIC stream: Application error 0x0" connIndex=3 ip=198.41.192.57
2022-07-25T11:46:03Z WRN Serve tunnel error error="failed to accept QUIC stream: Application error 0x0" connIndex=3 ip=198.41.192.57
2022-07-25T11:46:03Z INF Retrying connection in up to 1s seconds connIndex=3 ip=198.41.192.57
2022-07-25T11:46:04Z WRN Connection terminated error="failed to accept QUIC stream: Application error 0x0" connIndex=3
2022-07-25T11:46:15Z WRN Failed to create new quic connection error="failed to dial to edge with quic: timeout: no recent network activity" connIndex=3 ip=198.41.192.57
2022-07-25T11:46:15Z INF Retrying connection in up to 4s seconds connIndex=3 ip=198.41.192.57
2022-07-25T11:46:17Z WRN If this log occurs persistently, and cloudflared is unable to connect to Cloudflare Network with `quic` protocol, then most likely your machine/network is getting its egress UDP to port 7844 (or others) blocked or dropped. Make sure to allow egress connectivity as per https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/ports-and-ips/
If you are using private routing to this Tunnel, then UDP (and Private DNS Resolution) will not work unless your cloudflared can connect with Cloudflare Network with `quic`. connIndex=3 ip=198.41.192.57
2022-07-25T11:46:17Z INF Switching to fallback protocol http2 connIndex=3 ip=198.41.192.57
2022-07-25T11:46:17Z WRN Connection terminated error="failed to dial to edge with quic: timeout: no recent network activity" connIndex=3
2022-07-25T11:46:51Z INF Connection 5487d148-aa1b-43a2-a79e-a69026bfb218 registered connIndex=3 ip=198.41.192.57 location=FRA
2022-07-25T12:05:55Z INF Unregistered tunnel connection connIndex=1
2022-07-25T12:05:55Z WRN Failed to serve quic connection error="failed to accept QUIC stream: Application error 0x0" connIndex=1 ip=198.41.192.107
2022-07-25T12:05:55Z WRN Serve tunnel error error="failed to accept QUIC stream: Application error 0x0" connIndex=1 ip=198.41.192.107
2022-07-25T12:05:55Z INF Retrying connection in up to 1s seconds connIndex=1 ip=198.41.192.107
2022-07-25T12:05:57Z WRN Connection terminated error="failed to accept QUIC stream: Application error 0x0" connIndex=1
2022-07-25T12:06:07Z INF Connection f80e8a53-d6a9-45aa-a32b-6894da757542 registered connIndex=1 ip=198.41.192.107 location=FRA
2022-07-25T12:46:45Z INF abort: rpc: aborted by remote: rpc: shutdown
2022-07-25T12:46:45Z INF Unregistered tunnel connection connIndex=0
2022-07-25T12:46:45Z WRN Failed to serve quic connection error="failed to accept QUIC stream: Application error 0x0" connIndex=0 ip=198.41.200.193
2022-07-25T12:46:45Z WRN Serve tunnel error error="failed to accept QUIC stream: Application error 0x0" connIndex=0 ip=198.41.200.193
2022-07-25T12:46:45Z INF Retrying connection in up to 1s seconds connIndex=0 ip=198.41.200.193
2022-07-25T12:46:46Z WRN Connection terminated error="failed to accept QUIC stream: Application error 0x0" connIndex=0
2022-07-25T12:47:05Z WRN Failed to create new quic connection error="failed to dial to edge with quic: timeout: no recent network activity" connIndex=0 ip=198.41.200.193
2022-07-25T12:47:05Z INF Retrying connection in up to 4s seconds connIndex=0 ip=198.41.200.193
2022-07-25T12:47:08Z WRN If this log occurs persistently, and cloudflared is unable to connect to Cloudflare Network with `quic` protocol, then most likely your machine/network is getting its egress UDP to port 7844 (or others) blocked or dropped. Make sure to allow egress connectivity as per https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/ports-and-ips/
If you are using private routing to this Tunnel, then UDP (and Private DNS Resolution) will not work unless your cloudflared can connect with Cloudflare Network with `quic`. connIndex=0 ip=198.41.200.193
2022-07-25T12:47:08Z INF Switching to fallback protocol http2 connIndex=0 ip=198.41.200.193
2022-07-25T12:47:08Z WRN Connection terminated error="failed to dial to edge with quic: timeout: no recent network activity" connIndex=0
2022-07-25T12:47:31Z WRN Unable to establish connection with Cloudflare edge error="TLS handshake with edge error: read tcp 172.30.33.5:37452->198.41.200.193:7844: i/o timeout" connIndex=0 ip=198.41.200.193
2022-07-25T12:47:31Z WRN Serve tunnel error error="TLS handshake with edge error: read tcp 172.30.33.5:37452->198.41.200.193:7844: i/o timeout" connIndex=0 ip=198.41.200.193
2022-07-25T12:47:31Z INF Retrying connection in up to 1s seconds connIndex=0 ip=198.41.200.193
2022-07-25T12:47:32Z WRN Connection terminated error="TLS handshake with edge error: read tcp 172.30.33.5:37452->198.41.200.193:7844: i/o timeout" connIndex=0
2022-07-25T12:47:53Z INF Connection e82d52d9-6c02-4df0-9b8a-414f776c8f30 registered connIndex=0 ip=198.41.200.193 location=AMS
2022-07-25T13:43:46Z INF Unregistered tunnel connection connIndex=1
2022-07-25T13:43:46Z WRN Failed to serve quic connection error="failed to accept QUIC stream: Application error 0x0" connIndex=1 ip=198.41.192.107
2022-07-25T13:43:46Z WRN Serve tunnel error error="failed to accept QUIC stream: Application error 0x0" connIndex=1 ip=198.41.192.107
2022-07-25T13:43:46Z INF Retrying connection in up to 1s seconds connIndex=1 ip=198.41.192.107
2022-07-25T13:43:47Z WRN Connection terminated error="failed to accept QUIC stream: Application error 0x0" connIndex=1
2022-07-25T13:43:49Z INF Connection 5abded0c-e165-4035-aa90-fa72ee61fdec registered connIndex=1 ip=198.41.192.107 location=FRA
2022-07-25T14:33:18Z INF Unregistered tunnel connection connIndex=2
2022-07-25T14:33:18Z WRN Failed to serve quic connection error="failed to accept QUIC stream: Application error 0x0" connIndex=2 ip=198.41.200.43
2022-07-25T14:33:18Z WRN Serve tunnel error error="failed to accept QUIC stream: Application error 0x0" connIndex=2 ip=198.41.200.43
2022-07-25T14:33:18Z INF Retrying connection in up to 1s seconds connIndex=2 ip=198.41.200.43
2022-07-25T14:33:18Z WRN Connection terminated error="failed to accept QUIC stream: Application error 0x0" connIndex=2
2022-07-25T14:33:25Z INF Connection 3a6d181b-568c-4fa9-8ac4-acad987e3041 registered connIndex=2 ip=198.41.200.43 location=AMS
2022-07-25T14:41:28Z INF Unregistered tunnel connection connIndex=3
2022-07-25T14:41:29Z INF Lost connection with the edge connIndex=3
2022-07-25T14:41:29Z WRN Serve tunnel error error="connection with edge closed" connIndex=3 ip=198.41.192.57
2022-07-25T14:41:29Z INF Retrying connection in up to 1s seconds connIndex=3 ip=198.41.192.57
2022-07-25T14:41:29Z INF Changing protocol to quic connIndex=3 ip=198.41.192.57
2022-07-25T14:41:29Z WRN Connection terminated error="connection with edge closed" connIndex=3
2022-07-25T14:41:30Z INF Connection 58cf3785-48d9-4997-a7b2-09d11e830346 registered connIndex=3 ip=198.41.192.57 location=FRA
2022-07-25T14:44:32Z INF Unregistered tunnel connection connIndex=0
2022-07-25T14:44:33Z INF Lost connection with the edge connIndex=0
2022-07-25T14:44:33Z WRN Serve tunnel error error="connection with edge closed" connIndex=0 ip=198.41.200.193
2022-07-25T14:44:33Z INF Retrying connection in up to 1s seconds connIndex=0 ip=198.41.200.193
2022-07-25T14:44:34Z INF Changing protocol to quic connIndex=0 ip=198.41.200.193
2022-07-25T14:44:34Z WRN Connection terminated error="connection with edge closed" connIndex=0
2022-07-25T14:44:51Z INF Connection 61afa2d0-5b6d-4b89-a3a4-273d5a6eed08 registered connIndex=0 ip=198.41.200.193 location=AMS

**Comment by elcajon** [comment]
> If I open enter my domain name in the browser, I get a 404 error and the log shows:

Take a look here https://github.com/brenner-tobias/addon-cloudflared/blob/main/cloudflared/DOCS.md#troubleshooting
This should fix the error. 

The log looks okay, there always are some warnings about tunnel failures.
